CruCentralCoast/Events/EventDetailsVC.swift

```diff
@@ -49,44 +57,113 @@ class EventDetailsVC: UIViewController {
+            let regionSpan = MKCoordinateRegionMakeWithDistance(coordinates, regionDistance, regionDistance)
+            let options = [
+                MKLaunchOptionsMapCenterKey: NSValue(mkCoordinate: regionSpan.center),
+                MKLaunchOptionsMapSpanKey: NSValue(mkCoordinateSpan: regionSpan.span)
+            ]
+            let mkPlacemark = MKPlacemark(coordinate: coordinates, addressDictionary: nil)
+            let mapItem = MKMapItem(placemark: mkPlacemark)
+            mapItem.name = self.eventViewableLocation
+            mapItem.openInMaps(launchOptions: options)
+        }
+    }
+    
+//    func getCoordinate(eventLocation: String) -> (CLLocationDegrees, CLLocationDegrees){
+//
+//        return (lat, lon)
+//    }
+    
+    //found at https://www.hackingwithswift.com/read/32/3/how-to-use-sfsafariviewcontroller-to-browse-a-web-page
     @IBAction func facebookButtonPressed(_ sender: Any) {
-        self.presentAlert(title: "Link not set up yet!!", message: "Coming soon")
+        guard let facebookURL = event?.facebookURL else {
+            return
+        }
+        
+        if let url = URL(string: facebookURL) {
+            let config = SFSafariViewController.Configuration()
+            config.entersReaderIfAvailable = true
+            
+            let vc = SFSafariViewController(url: url, configuration: config)
+            present(vc, animated: true)
+        }
     }
 
-    /* future function for calendar button on events Details Screen
-    @IBAction func calendarButtonPressed(_ sender: Any) {
+    @IBAction func addToCalendarButtonPressed(_ sender: Any) {
         let eventStore : EKEventStore = EKEventStore()
         
-        eventStore.requestAccess(to: .event, completion: { (granted, error) in
+        eventStore.requestAccess(to: .event) { (granted, error) in
             if (granted) && (error == nil) {
-                let event = EKEvent(eventStore: eventStore)
-                event.title = self.titleLabel.text
+                let event:EKEvent = EKEvent(eventStore: eventStore)
+                event.title = self.event?.title
+                event.startDate = self.event?.startDate
+                event.endDate = self.event?.endDate
+                event.notes = self.event?.description
+                event.calendar = eventStore.defaultCalendarForNewEvents
+                do {
+                    try eventStore.save(event,span: .thisEvent)
+                        let alert = UIAlertController(title: "Calendar", message: "Event Successfully added to calendar", preferredStyle: .alert)
+                        alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+                        self.present(alert, animated: true)
+                } catch let error as NSError {
+                    print("Error : \(error)")
+                }
+                print("Save Event")
+            } else {
+                print("error : \(String(describing: error))")
```

Comment:
I don't think you need the String(describing: error) part. It should automatically do that when printing an error